src/collector/metrics-collector.ts

```diff
@@ -102,21 +164,73 @@ export class MetricsCollector {
 	}
 
 	public async updateAll(): Promise<void> {
-		const updatePromises = this.queues.map((q) => getStats(q.prefix, q.name, q.queue, this.guages));
+		this.logger.debug('Starting metrics update for', this.queues.length, 'queues');
+		const updatePromises = this.queues.map(async (q) => {
+			try {
+				this.logger.debug('Updating metrics for queue:', q.name);
+				await getStats(q.prefix, q.name, q.queue, this.guages);
+				this.logger.debug('Metrics updated for queue:', q.name);
+			} catch (error) {
+				this.logger.error('Failed to update metrics for queue:', q.name, error);
+				throw error;
+			}
+		});
 		await Promise.all(updatePromises);
+		this.logger.debug('All metrics updated successfully');
 	}
 
 	public async ping(): Promise<void> {
-		await this.defaultRedisClient.ping();
+		try {
+			this.logger.debug('Sending Redis ping...');
+			await this.defaultRedisClient.ping();
+			this.logger.debug('Redis ping successful');
+		} catch (error) {
+			this.logger.error('Redis ping failed:', error);
+			throw error;
+		}
 	}
 
 	public async close(): Promise<void> {
-		this.defaultRedisClient.disconnect();
+		this.logger.info('Starting close process...');
+
+		this.logger.info('Removing event listeners...');
 		for (const q of this.queues) {
 			for (const l of this.myListeners) {
 				q.queueEvents.removeListener('completed', l);
 			}
 		}
-		await Promise.all(this.queues.reduce((ary, q) => ary.concat([q.queue.close(), q.queueEvents.close()]), [] as Promise<void>[]));
+		this.logger.info('Event listeners removed.');
+
+		// Force disconnect all Redis connections immediately
+		this.logger.info('Disconnecting main Redis client...');
+		this.defaultRedisClient.disconnect();
+		this.logger.info('Main Redis client disconnected.');
+
+		this.logger.info('Closing queues and queue events...');
+		const closePromises = this.queues.map(async (q) => {
+			try {
+				const queueClosePromise = q.queue.close();
+				const queueTimeoutPromise = new Promise((_, reject) =>
+					setTimeout(() => reject(new Error('Queue close timeout')), 500)
```

Comment:
do we really need all of these individual timeouts for queues and events if there's a higher level timeout around `collector.close()` already?